policy-controller/k8s/index/src/cluster_info.rs

```diff
@@ -0,0 +1,55 @@
+use std::num::NonZeroU16;
+
+use crate::{pod::PortSet, DefaultPolicy};
+use linkerd_policy_controller_core::IpNet;
+use tokio::time;
+
+/// Holds cluster metadata.
+#[derive(Clone, Debug)]
+pub struct ClusterInfo {
+    /// Networks including PodIPs in this cluster.
+    ///
+    /// Unfortunately, there's no way to discover this at runtime.
+    pub networks: Vec<IpNet>,
+
+    /// The namespace where the linkerd control plane is deployed
+    pub control_plane_ns: String,
+
+    /// E.g. "cluster.local"
+    pub dns_domain: String,
+
+    /// The cluster's mesh identity trust domain.
+    pub identity_domain: String,
+
+    /// The cluster-wide default policy.
+    pub default_policy: DefaultPolicy,
+
+    /// The cluster-wide default protocol detection timeout.
+    pub default_detect_timeout: time::Duration,
+
+    /// The default set of ports to be marked opaque.
+    pub default_opaque_ports: PortSet,
+
+    /// The networks that probes are expected to be from.
+    pub probe_networks: Vec<IpNet>,
+}
+
+impl ClusterInfo {
+    pub(crate) fn service_account_identity(&self, ns: &str, sa: &str) -> String {
+        format!(
+            "{}.{}.serviceaccount.identity.{}.{}",
+            sa, ns, self.control_plane_ns, self.identity_domain
+        )
+    }
+
+    pub(crate) fn namespace_identity(&self, ns: &str) -> String {
+        format!(
+            "*.{}.serviceaccount.identity.{}.{}",
+            ns, self.control_plane_ns, self.identity_domain
+        )
```

Comment:
I know this is just moving code around but we could use the other function here. Certainly not a blocker.

```suggestion
        self.service_account_identity(ns, "*")
```